perun/postprocess/moving_average/run.py

```diff
@@ -127,41 +139,53 @@ def simple_moving_median(ctx, **kwargs):
-            - **alpha**: specify smoothing factor :math:`{\\alpha}` directly: 0 < :math:`{\\alpha}` <= 1
-
-        The computed coefficient :math:`{\\alpha}` represents the degree of weighting decrease, a constant smoothing
-        factor, The higher value of :math:`{\\alpha}` discounts older observations faster, the small value to the
-        contrary. Finally, to calculate the current value of EMA is used the relevant formula. It is important
-        do not confuse **Exponential** Moving Average with **Simple** Moving Average. An **Exponential** Moving
-        Average behaves quite differently from the second mentioned method, because it is the function of
-        weighting factor or length of the average.
+        This method is a type of moving average methods, also know as **Exponential** Weighted
+        Moving Average, that places a greater weight and significance on the most recent data
+        points. The weighting for each far x-coordinate decreases exponentially and never reaching
+        zero. This approach of moving average reacts more significantly to recent changes than a
+        *Simple* Moving Average, which applies an equal weight to all observations in the period.
+        To calculate an EMA must be first computing the **Simple** Moving Average (SMA) over a
+        particular sub-interval. In the next step must be calculated the multiplier for smoothing
+        (weighting) the EMA, which depends on the selected formula, the following options are
+        supported (`<decay>`):
+
+            | - **com**: specify decay in terms of center of mass:
+                    :math:`{\\alpha}` = 1 / (1 + com), for com >= 0
+            | - **span**: specify decay in terms of span:
+                    :math:`{\\alpha}` = 2 / (span + 1), for span >= 1
+            | - **halflife**: specify decay in terms of half-life,
+                    :math:`{\\alpha}` = 1 - exp(log(0.5) / halflife), for halflife > 0
+            | - **alpha**: specify smoothing factor
+                    :math:`{\\alpha}` directly: 0 < :math:`{\\alpha}` <= 1
+
+        The computed coefficient :math:`{\\alpha}` represents the degree of weighting decrease, a
+        constant smoothing factor, The higher value of :math:`{\\alpha}` discounts older
+        observations faster, the small value to the contrary. Finally, to calculate the current
+        value of EMA is used the relevant formula. It is important do not confuse **Exponential**
+        Moving Average with **Simple** Moving Average. An **Exponential** Moving Average behaves
+        quite differently from the second mentioned method, because it is the function of weighting
+        factor or length of the average.
     """
-    kwargs.update({'moving_method': 'ema', 'window_width': kwargs['decay'][1], 'decay': kwargs['decay'][0]})
+    kwargs.update({
+        'moving_method': 'ema',
+        'window_width': kwargs['decay'][1],
```

Comment:
Suggestion: This may be used on several places, so maybe it would be worthy to extract to namedtuple, like `Decay`

Reply:
I think that is no needed because there is single direct access to items of these parameters and further in the code are used only items of the dictionary. Moreover, the item `window_width` is uniform at all moving-average methods and this change would probably run the wave of the other changes.

Reply:
Erm, I'm talking only about `kwargs['decay'], which currently contains tuple, and it could be instead namedtuple, so you would write `kwargs['decay'].rate` or something like that, I don't know what is in `[0]` and in `[1]`. Just consider if `kwargs['decay'][X]` is used in more places.

Reply:
Change to the `NamedTuple` would mislead the `click`, because now is used the type of this argument `click.Tuple` and moreover `kwargs['decay]` is used only in this place and further are used only `window_width` and `decay` items.

Reply:
AFAIK you can use in click custom classes as well, but if it is used only in this place, then it would lead to more changes and problem. So keep it like this then.